File: src/main/java/com/elroykanye/sadues/business/service/i/AcademicYearService.java
package com.elroykanye.sadues.business.service.i;

import com.elroykanye.sadues.api.dto.AcademicYearDto;
import com.elroykanye.sadues.api.dto.response.SaResponse;
import com.elroykanye.sadues.data.entity.AcademicYear;
import java.util.List;

public interface AcademicYearService {
    
    SaResponse create( AcademicYearDto dto);
    AcademicYear getEntity(long id);
    AcademicYearDto getDto(long id);
    List<AcademicYear> getAllEntities();
    List<AcademicYearDto> getAllDto();
    SaResponse update( AcademicYearDto dto);
    void delete(long id);
}

File: src/main/java/com/elroykanye/sadues/business/service/i/AcademicYearService.kt
package com.elroykanye.sadues.business.service

import com.elroykanye.sadues.api.dto.AcademicYearDto
import com.elroykanye.sadues.api.dto.response.SaResponse
import org.springframework.http.ResponseEntity

interface AcademicYearService {
    fun create(academicYearDto: AcademicYearDto): ResponseEntity<SaResponse> {
        TODO("Not yet implemented")
    }

    fun getDto(id: Long): ResponseEntity<AcademicYearDto> {
        TODO("Not yet implemented")
    }
    
    fun getAllDto(): ResponseEntity<List<AcademicYearDto>> {
        TODO("Not yet implemented")
    }

    fun update(academicYearDto: AcademicYearDto): ResponseEntity<SaResponse> {
        TODO("Not yet implemented")
    }

    fun delete(id: Long): ResponseEntity<Void> {
        TODO("Not yet implemented")
    }
} 
